### PR TITLE
Support for Ruby 2.0 and 2.1 has ended.

### PR DESCRIPTION
**Pull request by AlexWayfer**
https://www.ruby-lang.org/en/news/2016/02/24/support-plan-of-ruby-2-0-0-and-2-1/
https://www.ruby-lang.org/en/news/2017/04/01/support-of-ruby-2-1-has-ended/

Switch to pre-compiled patch-versions of Ruby.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,6 @@ branches:
 script:
   - bundle exec rake
 rvm:
-  - 2.0.0-p648
-  - 2.1.9
-  - 2.2.5
-  - 2.3.1
+  - 2.2
+  - 2.3
+  - 2.4
```
Comment (panthomakos) [inline .travis.yml]: What version is chosen when `2.2` is selected vs `2.2.5`?

**Comment by AlexWayfer** [comment]
> Although 2.0 and 2.1 are no longer fully supported, the gem works fine on both versions. I would prefer to retain the testing matrix unless there is another reason to drop support as well.

It looks meaningless: you can delete version support when something goes wrong, but what's the point then to test it until this point, if the sense of the tests is correcting the errors. And fixing errors for unsupported and unsafe old versions of Ruby looks also unreasonable.

File: .travis.yml (hunk 1 above)
Comment (AlexWayfer) [reply .travis.yml]: As I understand, pre-installed Ruby versions in Travis CI. [Some documentation](https://docs.travis-ci.com/user/languages/ruby/#Specifying-Ruby-versions-and-implementations). And in my experience it's close to the last released patch-version (with a slight lag). I don't see the need to specify the patch-version until there is something specific (bugs in the pre-installed version of Ruby in Travis CI, for example: bug in 2.4.1, Travis CI has 2.4.1 pre-installed for `2.4`, but `2.4.2` is already available).

**Comment by panthomakos** [comment]
I would prefer to not drop 2.0 and 2.1 support until a major release and until that time I would like to know if a change breaks tests for either of these versions. If tests do break due to a new feature addition, we can assess if we also want to drop 2.0 and 2.1 support at that time.

**Comment by AlexWayfer** [comment]
> If you can add 2.0 and 2.1 back, the rest of the changes look good to me.

OK, done…

**Pull request merged**